pydtk/models/movie.py

```diff
@@ -39,47 +39,55 @@ def _load(self, path, start_timestamp=None, end_timestamp=None, **kwargs):
         if not os.path.isfile(path):
             raise IOError('No such file: {}'.format(path))
 
-        frames = []
-        timestamps = []
-
+        # Get video info
         cap = cv2.VideoCapture(path)
+        assert cap.isOpened(), f"{path} cannot be opened!"
         fps = cap.get(cv2.CAP_PROP_FPS)
-        width = None
-        height = None
-        n_channels = None
+        n_frames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+        height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+        if int(cap.get(cv2.CAP_PROP_CHANNEL)) == 0:
+            _, frame = cap.read()  # Read the first frame to get n_channel
+            n_channels = frame.shape[2]
+        else:
+            n_channels = int(cap.get(cv2.CAP_PROP_CHANNEL))
+
+        # time[sec] -> frame index
+        if start_timestamp is not None:
+            start_frame_idx = np.floor(start_timestamp * fps).astype(int)
+        else:
+            start_frame_idx = 0
+        if end_timestamp is not None:
+            end_frame_idx = min(np.ceil(end_timestamp * fps).astype(int), n_frames - 1)
+        else:
+            end_frame_idx = n_frames - 1
+        frame_size = end_frame_idx - start_frame_idx + 1
+        assert 0 <= start_frame_idx < end_frame_idx < n_frames, "Timestamp out of range!"
+
+        # Read video
+        cap.set(cv2.CAP_PROP_POS_FRAMES, start_frame_idx)
+        data = np.empty((frame_size, height, width, n_channels))
```

Comment:
`data` should have dtype of unsigned int8 ([default is float](https://numpy.org/doc/stable/reference/generated/numpy.empty.html)).

Reply:
Fixed the code!